src/components/heading/heading.style.js

```diff
@@ -0,0 +1,35 @@
+import styled from 'styled-components';
+import Icon from '../icon';
+import baseTheme from '../../style/themes/base';
+
+const StyledHeadingIcon = styled(Icon)`
+  color: ${({ theme }) => theme.colors.border};
+  cursor: pointer;
+  top: 50%;
+  z-index: 1002;
+  position: absolute;
+
+  .carbon-heading--has-divider & {
+    margin-top: -8px;
+  }
+
+  &, &.carbon-icon {
+    position: absolute;
+    display: block;
+    color: ${({ theme }) => theme.icon};
+  }
+  
+  &:before, &.carbon-icon:before {
```

Comment:
Do we need the `.carbon-icon` class name here?

Reply:
yeah.

Reply:
Sorry I mean can you not use the `${StyledIcon}` instead, I may be missing something?

Reply:
oh! I didn't realise carbon-icon is StyledIcon. Thanks, it is updated now!